### PR TITLE
Add promises support and standard lint

### PR DESCRIPTION
**Pull request by tinchoz49**
This PR adds a Promise-based option to lock and also for maintenance purpose I added standard lint.

### CODE REVIEW

**Comment by mafintosh** [comment]
Neat! Can you move this promises stuff to ./promises.js? Also delete the lock file

**Comment by tinchoz49** [comment]
Hey @mafintosh, done.

File: promises.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+var mutexify = require('.')
+
+var mutexifyPromise = function () {
+  var lock = mutexify()
+
+  var acquire = function () {
+    return new Promise((resolve) => {
+      lock((release) => {
+        resolve(release)
+      })
```
Comment (mafintosh) [inline promises.js:10]: isn't this the same as lock(resolve)?

Comment (tinchoz49) [reply promises.js:10]: yes, oh it's true

Comment (tinchoz49) [reply promises.js]: fixed

**Comment by mafintosh** [comment]
Congratulations! You passed the review!

Claim your price at one npm release.

**Comment by tinchoz49** [comment]
haha :rocket:

**Comment by tinchoz49** [comment]
Should we update the travis, for the next release?

**Comment by mafintosh** [comment]
1.3.0

**Comment by mafintosh** [comment]
@tinchoz49 ya feel free. I renamed promises -> promise cause that what we called it in my similar thunky module

**Comment by tinchoz49** [comment]
> @tinchoz49 ya feel free. I renamed promises -> promise cause that what we called it in my similar thunky module

Ah that's good to know if I do other PR with promise stuff

**Pull request merged**